src/main/java/carpetaddonsnotfound/mixins/FlowerPotBlock_PickFlowersFromPotsMixin.java

```diff
@@ -0,0 +1,58 @@
+package carpetaddonsnotfound.mixins;
+
+import carpetaddonsnotfound.CarpetAddonsNotFoundSettings;
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Blocks;
+import net.minecraft.block.FlowerPotBlock;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.ActionResult;
+import net.minecraft.util.Hand;
+import net.minecraft.util.hit.BlockHitResult;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraft.world.event.GameEvent;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+@Mixin(FlowerPotBlock.class)
+public abstract class FlowerPotBlock_PickFlowersFromPotsMixin {
+  @Inject(
+          method = "onUse",
+          at = @At(
+                  value = "TAIL"
+          ),
+          cancellable = true
+  )
+  private void onUse(BlockState state,
+                     World world,
+                     BlockPos pos,
+                     PlayerEntity player,
+                     Hand hand,
+                     BlockHitResult hit,
+                     CallbackInfoReturnable<ActionResult> cir) {
+    Block potBlock = state.getBlock();
+    Block content;
+    if (!CarpetAddonsNotFoundSettings.alwaysPickFlowersFromPots || !(potBlock instanceof FlowerPotBlock) ||
+        (content = ((FlowerPotBlock) potBlock).getContent()) == Blocks.AIR) {
+      cir.cancel();
```

Comment:
I’m sorry to keep bringing it up but when I said this wasn’t needed I meant the entire bit with `cir` wasn’t needed in this if statement.

Reply:
Wait but so i removed necesarry part of code and it still worked?
Get the cross!